testapi.pm

```diff
@@ -244,43 +246,50 @@ sub _check_backend_response {
 }
 
 sub _check_or_assert {
-    my ($mustmatch, $timeout, $check) = @_;
-    $timeout = bmwqemu::scale_timeout($timeout);
+    my ($mustmatch, $check, %args) = @_;
+    $args{timeout} = bmwqemu::scale_timeout($args{timeout});
 
     die "current_test undefined" unless $autotest::current_test;
 
-    my $rsp = query_isotovideo('check_screen', {mustmatch => $mustmatch, timeout => $timeout, check => $check});
-    # seperate function because it needs to call itself
-    return _check_backend_response($rsp, $check, $timeout, $mustmatch);
+    my $rsp = query_isotovideo('check_screen', {mustmatch => $mustmatch, check => $check, timeout => $args{timeout}, no_wait => $args{no_wait}});
+    # separate function because it needs to call itself
+    return _check_backend_response($rsp, $check, $args{timeout}, $mustmatch, $args{no_wait});
 }
 
 =head2 assert_screen
 
-  assert_screen($mustmatch [,$timeout]);
+  assert_screen($mustmatch [, [$timeout] | [timeout => $timeout]] [, no_wait => $no_wait]);
 
 Wait for needle with tag C<$mustmatch> to appear on SUT screen. C<$mustmatch>
 can be string or C<ARRAYREF> of string (C<['tag1', 'tag2']>). The maximum
 waiting time is defined by C<$timeout>. It is recommended to use a value lower
 than the default timeout only when explicitly needed. C<assert_screen> is not
 very suitable for checking performance expectations. Under the normal
 circumstance of the screen being shown this does not imply a longer waiting
-time as the method returns as soon as a successful needle match occured.
+time as the method returns as soon as a successful needle match occurred.
+
+Specify C<$no_wait> to run the screen check as fast as possible that is
+possibly more than once per second which is default. Select this to check a
+screen which can change in a range faster than 1-2 seconds not to miss the
+screen to check for.
 
 Returns matched needle or throws C<NeedleFailed> exception if $timeout timeout
 is hit. Default timeout is 30s.
 
 =cut
 
 sub assert_screen {
-    my ($mustmatch, $timeout) = @_;
-    $timeout //= $bmwqemu::default_timeout;
-    bmwqemu::log_call(mustmatch => $mustmatch, timeout => $timeout);
-    return _check_or_assert($mustmatch, $timeout, 0);
+    my ($mustmatch) = shift;
+    my $timeout;
+    $timeout = shift if (@_ % 2);
+    my %args = (@_, timeout => $timeout // $bmwqemu::default_timeout);
+    bmwqemu::log_call(mustmatch => $mustmatch, %args);
+    return _check_or_assert($mustmatch, 0, %args);
 }
 
 =head2 check_screen
 
-  check_screen($mustmatch [,$timeout]);
+  check_screen($mustmatch [, [$timeout] | [timeout => $timeout]] [, no_wait => $no_wait]);
```

Comment:
I don't think this code supports check_screen 'hallo', timeout => 1

My reading of the code implies it will use default_timeout in these cases - and you didn't cover this code either